lib/srv/desktop/tdp/png.go

```diff
@@ -0,0 +1,24 @@
+package tdp
+
+import "image/png"
+
+// PNGEncoder returns the encoder used
+// for PNG Frames.
+func PNGEncoder() *png.Encoder {
+	return &png.Encoder{
+		CompressionLevel: png.BestSpeed,
+		BufferPool:       &pool{},
+	}
+}
+
+// pool implements png.EncoderBufferPool,
+// allowing us to reuse encoding resources
+type pool struct {
+	b *png.EncoderBuffer
+}
+
+// all encoding happens in a single thread, so we don't
+// need anything as sophisticated as a sync.Pool here
```

Comment:
Perhaps this requirement should be documented on the users of `PNGEncoder` as well?

Reply:
Good call, done!